File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int ans = 0;
    public int solution(int[] numbers, int target) {
        dfs(0, numbers, target, 0);
        
        return ans;
    }
    
    public void dfs(int idx, int[] numbers, int target, int sum){
        if(idx == numbers.length){
            if(sum == target){
                ans++;
            }
            return;
        }
        
        dfs(idx + 1, numbers, target, sum + numbers[idx]);
        
        dfs(idx + 1, numbers, target, sum - numbers[idx]);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
cnt = 0
def solution(numbers, target):
    global cnt
    cnt = 0
    dfs(numbers, target, 0, 0)
    return cnt
#numbers 배열에서 어떤 값을 더한 경우 뺀경우로 계속 dfs하기

def dfs(numbers, target, idx, total):
    global cnt
    if idx == len(numbers):
        if total == target:
            cnt += 1
        return
    
    dfs(numbers, target, idx + 1, total + numbers[idx])
    
    dfs(numbers, target, idx + 1, total - numbers[idx])